test/jdk/java/io/Reader/ReadAll.java

```diff
@@ -66,12 +66,55 @@ public static void setup() throws IOException {
         int size = rnd.nextInt(2, 16386);
 
         int plen = PHRASE.length();
-        List<String> strings = new ArrayList<String>(size);
+        StringBuilder sb = new StringBuilder(plen);
+        List<String> strings = new ArrayList<>(size);
         while (strings.size() < size) {
             int fromIndex = rnd.nextInt(0, plen / 2);
             int toIndex = rnd.nextInt(fromIndex, plen);
-            strings.add(PHRASE.substring(fromIndex, toIndex));
+            String s = PHRASE.substring(fromIndex, toIndex);
+            sb.append(s);
+            int bound = toIndex - fromIndex;
+            if (bound > 0) {
+                int offset = bound/2;
+                int n = rnd.nextInt(0, bound);
+                for (int i = 0; i < n; i++) {
+                    String f = null;
+                    switch (rnd.nextInt(7)) {
+                    case 0 -> f = "";
+                    case 1 -> f = "\r";
+                    case 2 -> f = "\n";
+                    case 3 -> f = "\r\n";
+                    case 4 -> f = "\r\r";
+                    case 5 -> f = "\n\n";
+                    case 6 -> f = " ";
+                    }
+                    sb.insert(offset, f);
+                }
+            }
+            strings.add(sb.toString());
+            sb.setLength(0);
         }
+
+        String p4096  = PHRASE.repeat((4096 + plen - 1)/plen);
+        String p8192  = PHRASE.repeat((8192 + plen - 1)/plen);
+        String p16384 = PHRASE.repeat((16384 + plen - 1)/plen);
+
+        for (int i = 0; i < 64; i++) {
+            for (int j = 0; j < 32; j++) {
+                switch (rnd.nextInt(8)) {
+                case 0 -> sb.append("");
+                case 1 -> sb.append(" ");
+                case 2 -> sb.append("\n");
+                case 3 -> sb.append(PHRASE);
+                case 5 -> sb.append(p4096);
+                case 6 -> sb.append(p8192);
+                case 7 -> sb.append(p16384);
+                }
+            }
+            strings.add(sb.toString());
+            sb.setLength(0);
+        }
+
```

Comment:
Does one of these cases result in a *very very long line without a terminator"; something that would trigger the limits of ArraySupport.newLength at or near Integer.MAX_VALUE -  8?

Reply:
I don't know, but I doubt it. I will look into adding something.

Reply:
Yea, seems a waste to run it every time. I don't think we run any tests statistically *some of the time".
Maybe if it was limited to run only in *big* configurations, maybe a separate \@run with:

`@requires os.maxMemory >= 6g`

Reply:
The test fails with OOME in the CI even with the above `@requires` so I am _not_ changing it for now.